build_runner_core/lib/src/environment/io_environment.dart

```diff
@@ -34,17 +35,33 @@ class IOEnvironment implements BuildEnvironment {
 
   final PackageGraph _packageGraph;
 
-  IOEnvironment(this._packageGraph,
-      {bool? assumeTty, bool outputSymlinksOnly = false})
-      : _isInteractive = assumeTty == true || _canPrompt(),
-        _outputSymlinksOnly = outputSymlinksOnly,
-        reader = FileBasedAssetReader(_packageGraph),
-        writer = FileBasedAssetWriter(_packageGraph) {
-    if (_outputSymlinksOnly && Platform.isWindows) {
+  IOEnvironment._(this.reader, this.writer, this._isInteractive,
+      this._outputSymlinksOnly, this._packageGraph);
+
+  factory IOEnvironment(
+    PackageGraph packageGraph, {
+    bool? assumeTty,
+    bool outputSymlinksOnly = false,
+    bool lowResourcesMode = false,
```

Comment:
FYI

I noticed that this param is never passed, so batching is always enabled.

Nothing to do, I'll figure out what the right fix is--I'm refactoring the "readers" anyway, that's why I noticed :)

Thanks.